### ISSUE TITLE
Genome nexus calls firing sequentially

### ISSUE THREAD

**Issue by alisman** [cBioPortal/cbioportal#9967]
There are two very time intensive calls to genomenexus api on mutation tab of results view.  It seems the second waits for the first. @inodb thinks this is unecessary.  Can we find out why they fire sequentially?

https://triage.cbioportal.mskcc.org/results/mutations?cancer_study_list=az_msk_impact&tab_index=tab_visualize&case_set_id=az_msk_impact_all&Action=Submit&gene_list=KRAS%2CTP53&Z_SCORE_THRESHOLD=2.0&RPPA_SCORE_THRESHOLD=2.0&profileFilter=mutations%2Cstructural_variants%2Ccna&geneset_list=%20

![image](https://user-images.githubusercontent.com/186521/213289096-5d839011-3b55-48f3-9882-1afbefc67a4d.png)

**Comment by onursumer** [comment]
The second call doesn't seem to explicitly wait for the first one.

The first one is for cancer hotspot annotations, and it is immediately called when the results page is loaded. I think because that one is required for OncoPrint: https://github.com/cBioPortal/cbioportal-frontend/blob/8972fe14eed3742ec6494b40eaeee245024f2566/src/shared/lib/comparison/AnalysisStore.ts#L243

The second one is only required for the mutation table in the `Mutations` tab and its invocation is delayed until mutation mapper store gets initialized in `createMutationMapperStoreForSelectedGene`: https://github.com/cBioPortal/cbioportal-frontend/blob/8972fe14eed3742ec6494b40eaeee245024f2566/src/pages/resultsView/ResultsViewPageStore.ts#L3744

I think this is on purpose to prevent loading of annotated mutation data until we are actually on the Mutations tab. It is a bit confusing that chrome reports these two calls as being sequential.

So, in short, the first call is immediately fired even if we are not on the `Mutations` tab, while the second one is only fired if the `Mutations` tab is active.

**Comment by onursumer** [comment]
Looks like the `Mutations` tab waits for `mutationsByGene` to start rendering, and if we follow the nested `await`s, we end up with this dependency chain: 
`mutationsByGene -> mutationsReportByGene -> _filteredAndAnnotatedMutationsReport -> getMutationPutativeDriverInfo -> isHotspotForOncoprint -> indexedHotspotData`

We need to further investigate if we really need the hotspot data returned by `mutationsReportByGene` to render the `Mutations` tab.

**Comment by onursumer** [comment]
Mutations tab needs the hotspot data returned by `mutationsReportByGene` to properly report the driver VS vus counts in the filter panel.

![image](https://user-images.githubusercontent.com/3604198/213815402-caf321b1-e466-46b3-aae5-28413c8723d5.png)
